transport/x/grpc/outbound.go

```diff
@@ -118,41 +134,32 @@ func (o *Outbound) invoke(
 	if responseMD != nil {
 		callOptions = []grpc.CallOption{grpc.Trailer(responseMD)}
 	}
+	apiPeer, onFinish, err := o.peerChooser.Choose(ctx, request)
+	defer func() {
+		if onFinish != nil {
```

Comment:
onFinish must not be nil. The peer lists are allowed to return a singleton, but the end of request needs to be tracked regardless.